lib/puppet/provider/mount/parsed.rb

```diff
@@ -178,7 +178,21 @@ def to_line(result)
       end
     end
   else
-    record_line self.name, :fields => @fields, :separator => /\s+/, :joiner => "\t", :optional => optional_fields
+    record_line self.name, :fields => @fields, :separator => /\s+/, :joiner => "\t", :optional => optional_fields, :block_eval => :instance do
+      def pre_gen(record)
+        if !record[:options] || record[:options].empty?
+          if Facter.value(:kernel) == 'Linux'
+            record[:options] = 'defaults'
+          else
+            raise RuntimeError, "Mount[#{record[:name]}]: Field 'options' is required"
+          end
+        end
+        if !record[:fstype] || record[:fstype].empty?
+          raise RuntimeError, "Mount[#{record[:name]}]: Field 'fstype' is required"
```

Comment:
`self.fail` gave me `Error: /Stage[main]/Main/Mount[/vol0]: Could not evaluate: private method `fail' called for #<Puppet::Util::FileParsing::FileRecord:0x0000000413a638>`, `Puppet.err` did error out, but also updated the contents of `/etc/fstab`. `fail` does what I would expect, as does `raise RuntimeError`, in that it throws an error, and doesn't update `/etc/fstab`